### PR TITLE
TLS Everywhere adoption

### PR DESCRIPTION
**Pull request by xek**

### CODE REVIEW

File: docs_user/modules/openstack-backend_services_deployment.adoc [hunk 1]
```diff
@@ -130,6 +132,10 @@ oc set data secret/osp-secret "SwiftPassword=$SWIFT_PASSWORD"
 * Deploy OpenStackControlPlane. *Make sure to only enable DNS,
 MariaDB, Memcached, and RabbitMQ services. All other services must
 be disabled.*
+
+* If the source environment enables TLS Everywhere, modify
+spec:tls:endpoint:internal:enabled to true.
```
[the remark this answers is not in the document]
Comment (olliewalsh) [reply docs_user/modules/openstack-backend_services_deployment.adoc:137]: ... or replace with instruction on how to disable TLSe for adoption

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+* Make sure the previous Adoption steps have been performed successfully.
+* Make sure the backend services on the new deployment are not started yet.
+
+== Variables
+
+Define the shell variables used in the following steps. The values are illustrative and refer to a single node standalone director deployment. Use values that are correct for your environment:
+
+IPA_SSH="ssh -i ~/install_yamls/out/edpm/ansibleee-ssh-key-id_rsa root@192.168.122.100 podman exec -ti freeipa-server-container"
+
+In this example the FreeIPA instance is running on a separate host, in a container.
```
[the remark this answers is not in the document]
Comment (xek) [reply docs_user/modules/openstack-tls_everywhere.adoc:25]: @jistr I'm curious what you think? This is currently written in a way so that we can execute the steps manually on our test environment

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+== Transfer the CA certificate and key from FreeIPA to the cert-manager Operator
+
+=== Exporting from FreeIPA
+
+There are a couple of ways to do it:
+
+1. You can copy the certificate from the `/etc/pki/pki-tomcat/alias` directory. The file is named `ca.crt`.
```
[the remark this answers is not in the document]
Comment (xek) [reply docs_user/modules/openstack-tls_everywhere.adoc:33]: Maybe we can skip this line, since we don't really point the user to retrieve the cert from there, but execute an SSH command

File: docs_user/modules/openstack-tls_everywhere.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (jistr) [reply docs_user/modules/openstack-tls_everywhere.adoc:25]: Yea Ollie has a point but at the moment we have quite a lot of CRC-isms in the examples everywhere. Maybe we should be adding `ifeval`s for these examples as well to split them between upstream and downstream variants [1]. But i wouldn't block this patch on it.

[1] https://github.com/openstack-k8s-operators/data-plane-adoption/blob/d3951b133ba112911e6a6e98afd1e44b8c167d31/docs_user/modules/openstack-mariadb_copy.adoc?plain=1#L50

[the remark this answers is not in the document]
Comment (jistr) [reply docs_user/modules/openstack-tls_everywhere.adoc:25]: Fabricio created a lint for this https://github.com/openstack-k8s-operators/data-plane-adoption/pull/368 so we should indeed split this example into upstream (dev focused with install yamls) vs. downstream (user focused) value, otherwise it will fail the lint.

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+== Stop and disable certmonger, and remove the existing certificates
+
+The final step on this activity is to stop and disable the certmonger service, and remove all certificates managed by it. The code below accomplishes this task:
+
+[source, bash]
+----
+#!/bin/bash
+
+if sudo systemctl is-active certmonger.service; then
```
Comment (xek) [inline docs_user/modules/openstack-tls_everywhere.adoc:180]: This has to be run on all EDPM nodes, see an example here: https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/docs_user/modules/openstack-stop_remaining_services.adoc?plain=1#L76

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    sudo systemctl disable --now certmonger.service
+fi
+
+# If necessary, adjust the directory below accordingly.
+certs_directory=/var/lib/certmonger/requests/
+certs_to_stop=$(ls -1 $certs_directory)
+number_of_certs=$(ls -1 $certs_directory | wc -l)
+
+if [ $? != 0 ] || [ $number_of_certs = 0 ]; then
+    echo "No certificates to be removed."
```
Comment (xek) [inline docs_user/modules/openstack-tls_everywhere.adoc:193]: It's not removing the certificates, but tracking requests for the certificates

File: docs_user/modules/openstack-backend_services_deployment.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (vakwetu) [reply docs_user/modules/openstack-backend_services_deployment.adoc:143]: +1 to replace with instruction to disable TLSe when not adopting from a TLS-E environment.

File: docs_user/modules/openstack-backend_services_deployment.adoc [hunk 6]
```diff
@@ -141,6 +147,12 @@ metadata:
 spec:
   secret: osp-secret
   storageClass: local-storage
+  tls:
+    endpoint:
+      internal:
+        enabled: true
+      public:
+        enabled: true
```
[the remark this answers is not in the document]
Comment (vakwetu) [reply docs_user/modules/openstack-backend_services_deployment.adoc:161]: same comment as above -- we need instruction for disabling tls-e if not adopting from a tls-e enabled environment

File: docs_user/modules/openstack-edpm_adoption.adoc [hunk 7]
```diff
@@ -260,6 +263,7 @@ kind: OpenStackDataPlaneNodeSet
 metadata:
   name: openstack
 spec:
+  tlsEnabled: true
```
Comment (vakwetu) [inline docs_user/modules/openstack-edpm_adoption.adoc:266]: As commented above this will default to true,  So we want an instruction to disable in case we are adopting from a non-tls environment.

Comment (vakwetu) [reply docs_user/modules/openstack-edpm_adoption.adoc:266]: non tls-e that is.

File: docs_user/modules/openstack-planning.adoc [hunk 8]
```diff
@@ -204,6 +204,23 @@ are eligible to run the different OpenStack services.  Refer to the xref:node-se
...
+which provisions the certificates for services running on the host.
+
+The new Operator based deployment uses cert-manager to provision the certificates.
+Instructions will be provided on how to retrieve the root CA from IPA and copy it
+over to the new environment.
+
+Because the same root CA is used to generate new certs, the currently used trust
+chain doesn't have to be modified.
+
+Note: The asumption is, that the new deployment will adopt the settings from the
```
Comment (vakwetu) [inline docs_user/modules/openstack-planning.adoc:220]: nit: typo assumption

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+----
+IPA_SSH="ssh -i ~/install_yamls/out/edpm/ansibleee-ssh-key-id_rsa root@192.168.122.100 podman exec -ti freeipa-server-container"
+----
+
+In this example the FreeIPA instance is running on a separate host, in a container.
+
+== Transfer the CA certificate and key from FreeIPA to the cert-manager Operator
+
+=== Exporting from FreeIPA
+
```
Comment (vakwetu) [inline docs_user/modules/openstack-tls_everywhere.adoc:30]: This is only going to work if : 

1. The CA that is used for tls-e in osp 16/17 is not used for anything else.  In that case, it may be advisable not to extract the CA cert.  It is the expectation that this CA will be decommissioned afterwards and the new cert-manager CA will be used instead.
2. The IPA CA has its certs stored in nssdb.  Otherwise, if the cert is stored in an HSM, for instance, the key may not be retrievable.  Or if it is retrievable, we may require special HSM utilities.

File: docs_user/modules/openstack-tls_everywhere.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (vakwetu) [reply docs_user/modules/openstack-tls_everywhere.adoc:25]: +1 on using real host for customer docs.

File: docs_user/modules/openstack-planning.adoc [hunk 10]
```diff
@@ -204,6 +204,23 @@ are eligible to run the different OpenStack services.  Refer to the xref:node-se
...
 
+== TLS Everywhere
+
+If the Director deployment was deployed with TLS Everywhere, FreeIPA (IDM) is used
+for providing services with certificates. Certmonger is installed on all hosts,
+which provisions the certificates for services running on the host.
+
+The new Operator based deployment uses cert-manager to provision the certificates.
+Instructions will be provided on how to retrieve the root CA from IPA and copy it
+over to the new environment.
```
Comment (vakwetu) [inline docs_user/modules/openstack-planning.adoc:215]: Lets be a little more precise.

If the Director deployment was deployed with TLS Everywhere, FreeIPA (IdM) is used
to issue certificates for the OpenStack services. Certmonger, a client process which is installed on all hosts,
interacts with FreeIPA (IdM) to request, install, track and renew these certificates.

The new Operator based deployment uses the cert-manager operator to issue, track and renew the certificates.

File: docs_user/modules/openstack-planning.adoc (hunk 8 above)
Comment (vakwetu) [reply docs_user/modules/openstack-planning.adoc:220]: Also, I'd put this note as the first paragraph here.

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+Server-Cert cert-pki-ca                                      u,u,u
+subsystemCert cert-pki-ca                                    u,u,u
+auditSigningCert cert-pki-ca                                 u,u,Pu
+----
+
+The item you need to consider is the first one: `caSigningCert cert-pki-ca`.
+
+==== Exporting the items (certificate and key)
+
+The command below generates a P12 file with both, the certificate and the key. The `/etc/pki/pki-tomcat/alias/pwdfile.txt` file contains the password that protects the key. You can use it to both, extract the key and generate the new file, `/tmp/freeipa.p12`. You can also choose another password. Should you choose to apply a different password for the new file, simply replace the parameter of the `-w` option, or alternatively use the `-W` (capital W) option followed by the password (in clear text).
```
Comment (vakwetu) [inline docs_user/modules/openstack-tls_everywhere.adoc:61]: I would suggest that you instead just supply the example where you create a pk12 file with a different password.
Its generally not a good idea to reuse passwords like you have done here.

So - 1) create password file
        2) extract using pk12util

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+----
+KEY_LENGTH=`$IPA_SSH openssl pkcs12 -in /tmp/freeipa.p12 -passin file:/etc/pki/pki-tomcat/alias/pwdfile.txt -nocerts -noenc | openssl rsa -text -noout | awk -F'[^0-9]+' '{ print $2; exit }'`
+----
+
+==== Importing the Items and Provisioning the Custom Resources (CRs) on OpenShift
+
+You will have to create two YAML files. The first one has three sections:
+
+* The `openstack` namespace, which is used by other Red Hat OpenStack operators and CRs.
+* The issuer. The cert-manager operator makes use of a self-signed issuer to issue both, the CA certificate itself and other certificates.
```
Comment (vakwetu) [inline docs_user/modules/openstack-tls_everywhere.adoc:80]: nit - extra comma.  "issue both, the CA"  should be "issue both the CA"

File: docs_user/modules/openstack-tls_everywhere.adoc (hunk 11 above)
Comment (vakwetu) [reply docs_user/modules/openstack-tls_everywhere.adoc:61]: Ignore this last comment.  Looks like you never actually copy the pk12 file out of the IPA server.  Thats great.

File: tests/config/base/openstack_control_plane.yaml [hunk 13]
```diff
@@ -6,6 +6,27 @@ spec:
...
+      ca:
+        customIssuer: rootca-internal
+      enabled: true
+
+  barbican:
+    enabled: false
+    template:
+      barbicanAPI: {}
+      barbicanWorker: {}
+      barbicanKeystoneListener: {}
```
Comment (vakwetu) [inline tests/config/base/openstack_control_plane.yaml:28]: nit: this should be in a different PR, right?

Comment (vakwetu) [reply tests/config/base/openstack_control_plane.yaml:28]: that is - the barbican enablement.

File: tests/roles/dataplane_adoption/tasks/nova_ffu.yaml [hunk 14]
```diff
@@ -65,28 +65,14 @@
     apiVersion: v1
```
Comment (vakwetu) [inline tests/roles/dataplane_adoption/tasks/nova_ffu.yaml:65]: Note:  The changes in this file should not be included.  They were meant for testing only.

The changes that are needed here would be provided as part of https://github.com/openstack-k8s-operators/data-plane-adoption/pull/376 - which should be a depends-on for this patch.

File: docs_user/modules/openstack-tls_everywhere.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (vakwetu) [reply docs_user/modules/openstack-tls_everywhere.adoc:33]: agreed

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+
+----
+oc get issuers -n openstack
+----
+
+----
+oc get secret rootca-internal -n openstack -o yaml
+----
+
+== Stop and disable certmonger, and remove the existing certificates
```
[the remark this answers is not in the document]
Comment (mauricioharley) [reply docs_user/modules/openstack-tls_everywhere.adoc:172]: It has the same effect. I reached out to the FreeIPA guys and that was the information I received.

File: docs_user/modules/openstack-tls_everywhere.adoc (hunk 5 above)
Comment (mauricioharley) [reply docs_user/modules/openstack-tls_everywhere.adoc:193]: As ditto above, stopping to track a certificate has the same effect as removing it, according to the information provided by the FreeIPA team.

File: docs_user/modules/openstack-tls_everywhere.adoc (hunk 15 above)
[the remark this answers is not in the document]
Comment (vakwetu) [reply docs_user/modules/openstack-tls_everywhere.adoc:172]: I disagree.  Its not the same thing.  Usually, there are post-scripts that are triggered when the cert is issued that copy the cert to specific locations.  Removing the cert request does not undo that action.  The cert remains in place.

**Comment by vakwetu** [comment]
recheck

**Comment by vakwetu** [comment]
recheck

File: docs_user/modules/openstack-backend_services_deployment.adoc [hunk 16]
```diff
@@ -147,6 +154,12 @@ metadata:
 spec:
   secret: osp-secret
   storageClass: local-storage
+  tls:
+    endpoint:
+      internal:
+        enabled: true
```
Comment (stuggi) [inline docs_user/modules/openstack-backend_services_deployment.adoc:160]: internal changed to `podLevel`  https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/samples/core_v1beta1_openstackcontrolplane_network_isolation_tlse.yaml#L10:

```
  tls:
    podLevel:
      enabled: true
```

File: docs_user/modules/openstack-backend_services_deployment.adoc [hunk 17]
```diff
@@ -136,6 +138,11 @@ oc set data secret/osp-secret "SwiftPassword=$SWIFT_PASSWORD"
 * Deploy OpenStackControlPlane. *Make sure to only enable DNS,
 MariaDB, Memcached, and RabbitMQ services. All other services must
 be disabled.*
+
+* If the source environment does not use TLS Everywhere, modify
+spec:tls:endpoint:internal:enabled to false. Do the same for
```
Comment (stuggi) [inline docs_user/modules/openstack-backend_services_deployment.adoc:143]: see comment bellow

File: docs_user/modules/openstack-backend_services_deployment.adoc (hunk 16 above)
Comment (stuggi) [reply docs_user/modules/openstack-backend_services_deployment.adoc:160]: I see in tests/config/base/openstack_control_plane.yaml it is correct

File: tests/roles/tls_adoption/tasks/main.yaml [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+        secretName: rootca-internal
+    EOF
+
+- name: Stop certmonger service
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    EDPM_PRIVATEKEY_PATH="{{ edpm_privatekey_path }}"
+    declare -A computes
+    computes=(
+      ["standalone.localdomain"]="{{ edpm_node_ip }}"
```
Comment (fao89) [inline tests/roles/tls_adoption/tasks/main.yaml:38]:
```suggestion
      ["standalone.ooo.test"]="{{ edpm_node_ip }}"
```

Comment (xek) [reply tests/roles/tls_adoption/tasks/main.yaml]: Done

File: tests/vars.sample.yaml [hunk 19]
```diff
@@ -1,5 +1,9 @@
 # Path to install_yamls repo
 install_yamls_path: ~/install_yamls #CUSTOMIZE_THIS
+
+# This flag signifies if TLS Everywhere is enabled on the source cloud
+enable_tlse: false
```
Comment (fao89) [inline tests/vars.sample.yaml:5]: you need to add it here: https://github.com/rdo-infra/rdo-jobs/blob/master/playbooks/data_plane_adoption/vars.yaml

File: tests/playbooks/test_with_ceph.yaml [hunk 20]
```diff
@@ -16,6 +16,10 @@
     - role: development_environment
       tags:
         - development_environment
+    - role: tls_adoption
+      tags:
+        - tls_adoption
+      when: enable_tlse|bool
```
Comment (fao89) [inline tests/playbooks/test_with_ceph.yaml:22]:
```suggestion
      when: enable_tlse|default(false)
```

File: tests/playbooks/test_minimal.yaml [hunk 21]
```diff
@@ -12,6 +12,10 @@
     - role: development_environment
       tags:
         - development_environment
+    - role: tls_adoption
+      tags:
+        - tls_adoption
+      when: enable_tlse|bool
```
Comment (fao89) [inline tests/playbooks/test_minimal.yaml:18]:
```suggestion
      when: enable_tlse|default(false)
```

**Review by fao89 — COMMENTED** [review]
```
TASK [tls_adoption : Create Certificate Issuer with cert and key from IPA] *****
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'enable_tlse|bool' failed. The error was: error while evaluating conditional (enable_tlse|bool): 'enable_tlse' is undefined.
```
https://review.rdoproject.org/zuul/build/c4f277744a5149ec9841a2d64936ff99

File: tests/vars.sample.yaml (hunk 19 above)
Comment (xek) [reply tests/vars.sample.yaml:5]: Yep, I already created a review https://review.rdoproject.org/r/c/rdo-jobs/+/52626

**Comment by xek** [comment]
> ```
> TASK [tls_adoption : Create Certificate Issuer with cert and key from IPA] *****
> fatal: [localhost]: FAILED! => {"msg": "The conditional check 'enable_tlse|bool' failed. The error was: error while evaluating conditional (enable_tlse|bool): 'enable_tlse' is undefined.
> ```
> 
> https://review.rdoproject.org/zuul/build/c4f277744a5149ec9841a2d64936ff99

Thanks for helping out! I applied the changes.

File: docs_user/modules/openstack-tls_everywhere.adoc [hunk 22]
```diff
@@ -0,0 +1,221 @@
...
+=== Exporting from FreeIPA
+
+You will need to interact with the database located in the `/etc/pki/pki-tomcat/alias` directory ("*db" files).
+
+=== Locating the CA certificate and key
+
+List all the certificates inside your NSSDB:
+
+----
+$ $IPA_SSH certutil -L -d /etc/pki/pki-tomcat/alias
```
Comment (afaranha) [inline docs_user/modules/openstack-tls_everywhere.adoc:50]: We can remove the starting '$' we don't have it on the other instructions

**Comment by xek** [comment]
I noticed the ordering in the ToC is incorrect, the TLS Everywhere page should be before creating the openstack OpenStackControlPlane. This is probably because of the restructuring in the docs. I'll look into it.

**Comment by xek** [comment]
I adjusted the ToC and headers inside the TLS-E page.
@klgill maybe you could also take a look if this can go in as is.

**Review by xek — COMMENTED** [review]
I did a full review of the current set of patches. Today we did just some minor adjustments after off-github review and discussions.

**Review by klgill — APPROVED** [review]

**Comment by fao89** [comment]
recheck

**Comment by xek** [comment]
recheck

**Comment by xek** [comment]
recheck

File: Makefile [hunk 23]
```diff
@@ -11,6 +11,16 @@ test-minimal:
 	mkdir -p tests/logs
 	ANSIBLE_CONFIG=$(TEST_CONFIG) ansible-playbook -v -i $(TEST_INVENTORY) -e @$(TEST_VARS) -e @$(TEST_SECRETS) $(TEST_ARGS) tests/playbooks/test_minimal.yaml 2>&1 | tee $(TEST_OUTFILE)
 
+test-minimal-tls: TEST_OUTFILE := tests/logs/test_minimal_out_$(shell date +%FT%T%Z).log
+test-minimal-tls:
+	mkdir -p tests/logs
+	ANSIBLE_CONFIG=$(TEST_CONFIG) ansible-playbook -v -i $(TEST_INVENTORY) -e @$(TEST_VARS) -e @$(TEST_SECRETS) --tags tls_adoption tests/playbooks/test_minimal.yaml 2>&1 | tee $(TEST_OUTFILE)
```
Comment (jistr) [inline Makefile:17]: Do we need these targets? We can use the `TEST_ARGS` variable to provide `--tags`. Not a blocker for the initial merge though - if we want to get rid of these, it can be done afterwards.

File: docs_user/modules/con_tlse-description.adoc [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+old deployment, so in case TLS Everywhere is disabled, it won't be enabled on
+the new deployment.
+
+If the Director deployment was deployed with TLS Everywhere, FreeIPA (IdM) is used
+to issue certificates for the OpenStack services. Certmonger, a client process which
+is installed on all hosts, interacts with FreeIPA (IdM) to request, install, track
+and renew these certificates.
+
+The new Operator based deployment uses the cert-manager operator to issue, track
+and renew the certificates.
```
Comment (jistr) [inline docs_user/modules/con_tlse-description.adoc:15]: This raises a question in my mind: how does cert renewal work on data plane, as any Operator doesn't act on the data plane nodes directly? (Again, not a blocker.)

File: docs_user/modules/openstack-dataplane_adoption.adoc [hunk 25]
```diff
@@ -251,6 +251,9 @@ external_ids        : {hostname=standalone.localdomain, ovn-bridge=br-int, ovn-b
 ...
 ----
 In above example bridge mappings are set as "datacentre:br-ctlplane" and it has to be set in the OpenStackDataPlaneNodeSet CR also.
+
+* Ommit, or change spec:tlsEnabled below to true, if the source environment uses TLS Everywhere. (This setting defaults to true.)
```
Comment (jistr) [inline docs_user/modules/openstack-dataplane_adoption.adoc:255]: "Ommit" -> "omit", not a blocker.

File: docs_user/modules/proc_migrating-tls-everywhere.adoc [hunk 26]
```diff
@@ -0,0 +1,187 @@
...
+
+Define the map of compute node name, IP pairs.
+
+[subs=+quotes]
+----
+declare -A computes
+computes=(
+  ["standalone.localdomain"]="192.168.122.100"
+  # ...
+)
```
Comment (jistr) [inline docs_user/modules/proc_migrating-tls-everywhere.adoc:40]: This is an undo of what https://issues.redhat.com/browse/OSPRH-2301 achieved -- we don't want to ssh into compute nodes because there can be hundreds of them.

I don't think we should block the initial merge on this, but can we subsequently move this into edpm-ansible just like we moved everything for Nova? It looks like this is needed only for removal of certmonger and associated cleanup, so this should be quite easily doable with edpm-ansible hopefully?

**Review by jistr — APPROVED** [review]
Nice :+1: . LGTM for the initial merge. I commented inline on a few small things, and the "not SSHing into compute nodes" is important but i think it can also be done as a follow up.

File: docs_user/modules/proc_migrating-tls-everywhere.adoc (hunk 26 above)
Comment (jistr) [reply docs_user/modules/proc_migrating-tls-everywhere.adoc:40]: We could track this as another story under https://issues.redhat.com/browse/OSPRH-1488 and we could make verification of https://issues.redhat.com/browse/OSPRH-2301 depend on that story.

File: docs_user/modules/con_tlse-description.adoc (hunk 24 above)
Comment (xek) [reply docs_user/modules/con_tlse-description.adoc:15]: There is an ansible playbook that runs when the secret changes, which deploys the certificate from the control plane into the data plane.  That's covered under the dataplane-operator TLS-E implementation.

File: docs_user/modules/proc_migrating-tls-everywhere.adoc (hunk 26 above)
Comment (xek) [reply docs_user/modules/proc_migrating-tls-everywhere.adoc:40]: This makes sense, it's mostly just another service to stop, so it should be easy to integrate.

**Comment by xek** [comment]
recheck

**Comment by xek** [comment]
recheck

File: Makefile (hunk 23 above)
Comment (xek) [reply Makefile]: Right, I removed those

File: docs_user/modules/openstack-dataplane_adoption.adoc (hunk 25 above)
Comment (xek) [reply docs_user/modules/openstack-dataplane_adoption.adoc]: Fixed

**Comment by jistr** [comment]
@xek ok to merge this one and iterate?

**Comment by xek** [comment]
@jistr yep, go ahead

**Comment by marios** [comment]
this one broke the periodic job - see the last three builds [1] [2] [3] 

```
TASK [backend_services : run kustomize and create controlplane with container overrides] ***
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "set -euxo pipefail\n\n\nmkdir -p tmp\noc kustomize periodic_ci > tmp/test_deployment.yaml\noc apply -f tmp/test_deployment.yaml\n", "delta": "0:00:00.271966", "end": "2024-04-20 04:32:37.536389", "msg": "non-zero return code", "rc": 1, "start": "2024-04-20 04:32:37.264423", "stderr": "+ mkdir -p tmp\n+ oc kustomize periodic_ci\n# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.\n# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.\n# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.\nerror: trouble configuring builtin PatchStrategicMergeTransformer with config: `\npaths:\n- ../base/tls_overrides.yaml\n- container_image_overrides.yaml\n`: security; file '/home/zuul/src/github.com/openstack-k8s-operators/data-plane-adoption/tests/config/base/tls_overrides.yaml' is not in or below '/home/zuul/src/github.com/openstack-k8s-operators/data-plane-adoption/tests/config/periodic_ci'", "stderr_lines": ["+ mkdir -p tmp", "+ oc kustomize periodic_ci", "# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.", "# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.", "# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.", "error: trouble configuring builtin PatchStrategicMergeTransformer with config: `", "paths:", "- ../base/tls_overrides.yaml", "- container_image_overrides.yaml", "`: security; file '/home/zuul/src/github.com/openstack-k8s-operators/data-plane-adoption/tests/config/base/tls_overrides.yaml' is not in or below '/home/zuul/src/github.com/openstack-k8s-operators/data-plane-adoption/tests/config/periodic_ci'"], "stdout": "", "stdout_lines": []}

```

The patch at https://github.com/openstack-k8s-operators/data-plane-adoption/pull/415 for the 'wrong content' cix will also fix this though.

Moving forward lets try to sanity check the periodic/downstream jobs are not broken by these large changes that have a good chance of disruption. cc @jistr @cescgina 




[1] https://review.rdoproject.org/zuul/build/d60664b9729a48eaa46fb51e349763de
[2] https://review.rdoproject.org/zuul/build/b257531465f248d1b7dcec2c7ec95d5c 
[3] https://review.rdoproject.org/zuul/build/a6dfceebb7574ddeb9af35d9bc9e0370

File: docs_user/modules/con_tlse-description.adoc (hunk 24 above)
Comment (olliewalsh) [reply docs_user/modules/con_tlse-description.adoc:15]: > There is an ansible playbook that runs when the secret changes

nope, user must run a deployment
